### DISCUSSION TITLE
Could <string>.to_str return the string?

### DISCUSSION THREAD

**Discussion by jswelling** [zeroSteiner/rule-engine#76] [Ideas]
I'm dealing with a case where the incoming value could be either an integer or the string representation of an integer, e.g. "2".  I need to modify it with string operations.  This is easy enough if it is an integer, as I just apply the to_str method.  However, if the value came in as a string originally, .to_str appears to return None.  Is there any way I can set up <string>.to_str to simply return the string?  Shouldn't this be a default behavior?

**Comment by zeroSteiner** [comment]
Yeah most other languages behave in a similar manner by allowing coercion of a value to the type that it already is.

I might be able to get this done in the next few weeks but no promises.

**Reply by zeroSteiner** [reply]
I'm almost done now. I'd just like to implement propagation of type hints for compound types. Might be a another week.

**Reply by zeroSteiner** [reply]
This has been completed in the [4.2.0 release](https://github.com/zeroSteiner/rule-engine/releases/tag/v4.2.0).
